File: java/src/test/java/problems101_200/HouseRobberTest.java
package problems101_200;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class HouseRobberTest extends SampledTest {
    @Test
    public void test() {
        for (HouseRobber solution : HouseRobber.values()) {
            for (HouseRobberSample sample : getSamples(HouseRobberSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.rob(sample.input.nums))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems101_200/HouseRobberTest.kt
package problems101_200

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class HouseRobberTest : SampledTest() {
    @Test
    fun test() =
        HouseRobber.entries.forEach { solution ->
            getSamples<Array<HouseRobberSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.rob(input.nums))
                    .isEqualTo(output)
            }
        }
}
